src/lib.rs

```diff
@@ -121,25 +121,26 @@ pub fn consumer(working_dir: &PathBuf, source_dir: &Option<PathBuf>, result_map:
     let mut gcov_type = GcovType::Unknown;
 
     while let Some(work_item) = queue.pop() {
+        let mut result: Option<Vec<(String,CovResult)>> = None;
         let new_results = match work_item.format {
             ItemFormat::GCNO => {
                 let gcov_path = match work_item.item {
                     ItemType::Path(gcno_path) => {
                         if !is_llvm {
                             run_gcov(&gcno_path, branch_enabled, working_dir);
                         } else {
-                            call_parse_llvm_gcno(working_dir.to_str().unwrap(),
-                                                 gcno_path.parent().unwrap().join(gcno_path.file_stem().unwrap()).to_str().unwrap(),
-                                                 branch_enabled);
+                            result = Some(call_parse_llvm_gcno(working_dir.to_str().unwrap(),
```

Comment:
You can return here directly, so you don't need to have the Option result and you can clean up the rest of the code

Reply:
Or we could split the is_llvm part and the non llvm part, as they are getting very different